components/net/cookie_storage.rs

```diff
@@ -65,7 +78,61 @@ impl CookieStorage {
         }
 
         // Step 12
-        self.cookies.push(cookie);
+        let domain = reg_host(&cookie.cookie.domain.clone().unwrap_or("".to_owned())).unwrap_or("".to_owned());
+        if !self.cookies_map.contains_key(&domain) {
+        let tempdomain = domain.clone();
+            let cookies: Vec<Cookie> = Vec::new();
+            self.cookies_map.insert(tempdomain, cookies);
+        }
+        let mut cookies = self.cookies_map.get_mut(&domain).unwrap();
+
+        if cookies.len() == self.max_per_host {
+            // Step 12.1
```

Comment:
This code doesn't correspond with a step 12.1 from the spec, so this comment doesn't belong here.